docs/src/configuring-zed.md

```diff
@@ -1150,6 +1150,136 @@ To interpret all `.c` files as C++, files called `MyLockFile` as TOML and files
 }
 ```
 
+## Diagnostics
+
+- Description: Configuration for diagnostics-related features.
+- Setting: `diagnostics`
+- Default:
+
+```json
+{
+  "diagnostics": {
+    "include_warnings": true,
```

Comment:
* Can we use a diagnostics level here? 
LSP defines that there are 4 types of these: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnosticSeverity and a single boolean cannot cover them all.

I think it's better to be flexible and allow users to show all or almost none of them, if the want to.
We could also move that into `inline` part of the settings?

* If we do that, I wonder how useful `primary_only` would be: even now it seems redundant, have you found it useful?

I feel that we remove `use_rendered` unless you have a good reason to leave it be, if we remove `primary_only`, we'll have only 3 settings knobs: on/off, diagnostics level, and diagnostics interactivity.
That set looks quite small and good to me.

Reply:
To be clear, the `include_warnings` setting has nothing to do with this PR other than I have decided to just re-use the existing top level `diagnostics` key in the settings because I was tired of trying to come up with a better name.

For the level filtering part of this, I'm on board with the end goal, but I'm not focusing on it precisely for the same reasons I referenced with `-Werror`. For me, this is a "Thou shall not pass (while diagnostics are active)" sort of default. Allowing other folks to ignore diagnostics is fine, but its the sort of thing I'll focus on once we get to the polishing steps.

I'll discuss the `primary_only/use_rendered` stuff in separate response.